src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -73,45 +82,51 @@ public function __construct(
         $autoLastModified = true
     )
     {
-        $this->userId = $userId;
+        $this->userId = (string) $userId;
         $this->autoLastModified = $autoLastModified;
         $this->namespaces = $namespaces;
     }
 
     /**
-     * Process the given node.
+     * Process the given node and return eventual extra operations determined from the node.
      *
      * @param NodeInterface $node
+     *
+     * @return AddNodeOperation[] Additional operations that the client must execute for autocreated nodes.
      */
-    public function process(Node $node)
+    public function process(NodeInterface $node)
     {
         $this->validateNamespace($node->getName());
-        $this->additionalOperations = array();
 
         $nodeDef = $node->getPrimaryNodeType();
         $nodeTypes = $node->getMixinNodeTypes();
         array_unshift($nodeTypes, $nodeDef);
 
+        $additionalOperations = array();
         foreach ($nodeTypes as $nodeType) {
             /* @var $nodeType NodeTypeDefinitionInterface */
-            $this->processNodeWithType($node, $nodeType);
+            $additionalOperations = array_merge(
+                $additionalOperations,
+                $this->processNodeWithType($node, $nodeType)
+            );
         }
-    }
 
-    public function getAdditionalOperations()
```

Comment:
+1 I think. Only used in the tests and we now aggregate the additional operations locally in `process` from what I see.